experimental/displaynames/src/options.rs

```diff
@@ -0,0 +1,87 @@
+// This file is part of ICU4X. For terms of use, please see the file
+// called LICENSE at the top level of the ICU4X source tree
+// (online at: https://github.com/unicode-org/icu4x/blob/main/LICENSE ).
+
+//! Options for [`DisplayNames`](crate::DisplayNames).
+
+/// A bag of options defining how region codes will be translated by
+/// [`DisplayNames`](crate::DisplayNames).
+///
+/// # Example
+///
+/// ```
+/// use icu_displaynames::options;
+/// use icu_displaynames::DisplayNames;
+/// use icu_locid::Locale;
+///
+/// let locale = Locale::UND;
+/// let mut options: options::DisplayNamesOptions = Default::default();
+/// options.style = options::Style::Narrow;
+/// let display_name = DisplayNames::try_new_unstable(
+///     &icu_testdata::unstable(),
+///     &locale.into(),
+///     options,
+/// )
+/// .expect("Data should load successfully");
+///
+/// let region_code = "AE";
+/// assert_writeable_eq!(display_name.of(&region_code), "United Arab Emirates");
+#[derive(Debug, Eq, PartialEq, Clone, Default)]
+#[non_exhaustive]
+pub struct DisplayNamesOptions {
+    /// The formatting style to use for display name,
+    /// defaults to "long".
+    pub style: Style,
+    /// The fallback return when the system does not have the
+    /// requested display name, defaults to "code".
+    pub fallback: Fallback,
+    /// The language display kind, defaults to "dialect".
+    pub language_display: LanguageDisplay,
+}
+
+/// An enum for formatting style.
+#[allow(missing_docs)] // The variants are self explanotory.
+#[non_exhaustive]
+#[derive(Debug, Eq, PartialEq, Clone, Copy)]
+pub enum Style {
+    Narrow,
```

Comment:
Note: No Narrow data for regions, but there is Short data, and Narrow data maybe for other types

Reply:
There is no Long data in CLDR for territory displaynames. I am keeping these options as is for other types.